src/items.rs

```diff
@@ -408,6 +408,122 @@ impl<'a> FmtVisitor<'a> {
+            let next_span_start: BytePos = if i == enum_def.variants.len() - 1 {
+                span.hi
+            } else {
+                enum_def.variants[i + 1].span.lo
+            };
+
+            self.visit_variant(f, i == enum_def.variants.len() - 1, next_span_start);
+        }
+        self.block_indent -= config!(tab_spaces);
+
+        self.format_missing_with_indent(span.lo + BytePos(enum_snippet.rfind('}').unwrap() as u32));
+        self.changes.push_str_span(span, "}");
+    }
+
+    // Variant of a enum
+    fn visit_variant(&mut self,
+                     field: &ast::Variant,
+                     last_field: bool,
+                     next_span_start: BytePos)
+    {
+        if self.visit_attrs(&field.node.attrs) {
+            return;
+        }
+
+        if let ast::VariantKind::TupleVariantKind(ref types) = field.node.kind {
+            self.format_missing_with_indent(field.span.lo);
+
+            let vis = match field.node.vis {
+                ast::Visibility::Public => "pub ",
+                ast::Visibility::Inherited => ""
+            };
+
+            let mut header = vis.to_owned();
+            header.push_str(&field.node.name.to_string());
+            self.changes.push_str_span(field.span, &header);
+
+            let mut result = String::new();
+
+            if types.len() > 0 {
+                let comments = self.make_comments_for_list(Vec::new(),
+                                                           types.iter().map(|arg| arg.ty.span),
+                                                           ",",
+                                                           ")",
+                                                           |span| span.lo,
+                                                           |span| span.hi,
+                                                           next_span_start);
+
+                let type_strings: Vec<_> = types.iter()
+                                                .map(|arg| pprust::ty_to_string(&arg.ty))
+                                                .zip(comments.into_iter())
+                                                .collect();
+
+                result.push('(');
+
+                let indent = self.block_indent
+                             + vis.len()
+                             + field.node.name.to_string().len()
+                             + 1; // (
+
+                let budget = config!(ideal_width) - indent - 1; // )
+
+                let fmt = ListFormatting {
+                    tactic: ListTactic::HorizontalVertical,
+                    separator: ",",
+                    trailing_separator: SeparatorTactic::Never,
+                    indent: indent,
+                    h_width: budget,
+                    v_width: budget,
+                };
+                result.push_str(&write_list(&type_strings, &fmt));
+                result.push(')');
+            }
+
+            if let Some(ref expr) = field.node.disr_expr {
```

Comment:
How would we format such a thing would it go over budget? I think that case is extremely uncommon, since `disr_expr` is only set for C-style enums, which are generally short. I will add an assert.